oci/layout/oci_dest.go

```diff
@@ -177,8 +177,12 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 		return err
 	}
 
+	if d.ref.image == "" {
```

Comment:
Are you saying that this check should be in either `NewImageDestination` or in `newImageDestination`?

Reply:
Yes; by that time we already know that the upload will fail, so we can skip copying blobs if we report this to the user early.

(Both `ref.NewImageDestination` and `newImageDestination` work; `newImageDestination` is a bit better because it is more difficult to unintentionally bypass.)

Reply:
Put the check in `newImageDestinatioin`, fixed.